extensions/spotify-player/src/nowPlayingMenuBar.tsx

```diff
@@ -26,41 +25,37 @@ export default function NowPlayingMenuBar() {
   const [authorized, setAuthorized] = useState(false);
   const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    const fetchPlayerAndTrackState = async () => {
-      // Check if Spotify is installed (only try this once)
-      let isInstalled = spotifyInstalled;
-      if (isInstalled == undefined) {
-        isInstalled = await isSpotifyInstalled();
-        setSpotifyInstalled(await isSpotifyInstalled());
-      }
+  const fetchPlayerAndTrackState = async () => {
+    let result: [SpotifyState | null, TrackInfo | null] = [null, null];
 
-      setAuthorized(await isAuthorized());
-
-      // If Spotify is installed then fetch the player and track state
-      if (isInstalled) {
-        try {
-          const [state, track] = await Promise.all([getState(), getTrack()]);
-
-          setCurrentSpotifyState(state);
-          setCurrentlyPlayingTrack(track);
-
-          let newSubtitle: string | undefined;
-          if (track.id) {
-            newSubtitle = `${track.artist} – ${track.name}`;
-          }
-          await updateCommandMetadata({ subtitle: newSubtitle });
```

Comment:
Is this correct? If I understand the API correctly then the command metadata will only be updated when the now playing view command is executed (not in the background). That was my reasoning for keeping this code path in both the now playing view command AND the menu bar. 

In fact now that I think about it maybe the command metadata should only be updated in the menu bar command and not the view command since that will be come out of date unless the user interacts with it.

I wonder if a raycast API for updating other the command metadata for other commands in the extension would be useful here?

Reply:
>In fact now that I think about it maybe the command metadata should only be updated in the menu bar command and not the view command since that will be come out of date unless the user interacts with it.

yeah, makes sense. 
I removed the subtitle updating from the menu bar command since I didn't want to have it for both commands. 
I'm thinking now that probably it makes sense to have the subtitle updating in the menu bar command only and remove it from **"Now Playing"** 🤔

Reply:
> I'm thinking now that probably it makes sense to have the subtitle updating in the menu bar command only and remove it from "Now Playing" 🤔

I think this would make sense, I removed it from "Now Playing" for now.

Reply:
I brought it back